// @huddle01/web-core/dist/common-js/EnhancedSet.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;require('../chunk-4FMC7UH2.cjs');

// src/common-js/EnhancedSet.ts
var defaultCompareFn = (a, b) => {
  if (a < b) {
    return false;
  }
  if (a > b) {
    return true;
  }
  return false;
};
var EnhancedSet = (_class = class {
  
  
  __init() {this.getKey = (a, b) => {
    const key = this.compareFn(a, b) ? `${a}_${b}` : `${b}_${a}`;
    return key;
  }}
  get size() {
    return this.set.size;
  }
  __init2() {this.has = (a, b) => {
    const key = this.getKey(a, b);
    const value = this.set.has(key);
    return value;
  }}
  __init3() {this.add = (a, b) => {
    const key = this.getKey(a, b);
    this.set.add(key);
    return key;
  }}
  __init4() {this.delete = (a, b) => {
    const key = this.getKey(a, b);
    return this.set.delete(key);
  }}
  __init5() {this.clear = () => {
    this.set.clear();
  }}
  constructor(data) {;_class.prototype.__init.call(this);_class.prototype.__init2.call(this);_class.prototype.__init3.call(this);_class.prototype.__init4.call(this);_class.prototype.__init5.call(this);
    this.set = /* @__PURE__ */ new Set();
    if (data.compareFn)
      this.compareFn = data.compareFn;
    else
      this.compareFn = defaultCompareFn;
  }
}, _class);



exports.EnhancedSet = EnhancedSet; exports.defaultCompareFn = defaultCompareFn;
