// @huddle01/web-core/dist/chunk-3S34XHJL.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// src/validators/common.validators.ts
var _zod = require('zod');
var ConnectionParameterSchema = _zod.z.object({
  roomId: _zod.z.string({
    required_error: "Room id must provided"
  }),
  token: _zod.z.string({
    required_error: "Token must provided"
  })
});
var GeolocationSchema = _zod.z.object({
  country: _zod.z.string({
    required_error: "Country must provided"
  }),
  latitude: _zod.z.string({
    required_error: "Latitude must provided"
  }),
  longitude: _zod.z.string({
    required_error: "Longitude must provided"
  }),
  region: _zod.z.string({
    required_error: "Region must provided"
  }),
  globalRegion: _zod.z.string({
    required_error: "Global region must provided"
  }),
  ip: _zod.z.string({
    required_error: "IP must provided"
  })
});
var VolatileDataMessageSchema = _zod.z.object({
  from: _zod.z.string(),
  payload: _zod.z.unknown(),
  label: _zod.z.string().optional(),
  to: _zod.z.union([_zod.z.literal("*"), _zod.z.array(_zod.z.string())]).optional()
});
var DataMessageSchema = _zod.z.object({
  from: _zod.z.string(),
  payload: _zod.z.string(),
  label: _zod.z.string().optional()
});






exports.ConnectionParameterSchema = ConnectionParameterSchema; exports.GeolocationSchema = GeolocationSchema; exports.VolatileDataMessageSchema = VolatileDataMessageSchema; exports.DataMessageSchema = DataMessageSchema;
