// @huddle01/web-core/dist/chunk-QZEG6K3M.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;// src/common-js/EnhancedMap.ts
var defaultCompareFn = (a, b) => {
  if (a < b) {
    return false;
  }
  if (a > b) {
    return true;
  }
  return false;
};
var EnhancedMap = (_class = class {
  
  
  __init() {this.getKey = (a, b) => {
    const key = this.compareFn(a, b) ? `${a}_${b}` : `${b}_${a}`;
    return key;
  }}
  get size() {
    return this.map.size;
  }
  __init2() {this.get = (a, b) => {
    const key = this.getKey(a, b);
    const value = this.map.get(key);
    return value;
  }}
  __init3() {this.set = (a, b, value) => {
    const key = this.getKey(a, b);
    this.map.set(key, value);
    return value;
  }}
  __init4() {this.delete = (a, b) => {
    const key = this.getKey(a, b);
    return this.map.delete(key);
  }}
  __init5() {this.clear = () => {
    this.map.clear();
  }}
  constructor(data) {;_class.prototype.__init.call(this);_class.prototype.__init2.call(this);_class.prototype.__init3.call(this);_class.prototype.__init4.call(this);_class.prototype.__init5.call(this);
    this.map = /* @__PURE__ */ new Map();
    if (data.compareFn)
      this.compareFn = data.compareFn;
    else
      this.compareFn = defaultCompareFn;
  }
}, _class);




exports.defaultCompareFn = defaultCompareFn; exports.EnhancedMap = EnhancedMap;
